### PR TITLE
RTE Comment implementation 3

### PR DESCRIPTION
**Pull request by Milos-RTEi**

### CODE REVIEW

**Comment by Milos-RTEi** [comment]
I will deal with replacing "solution with map" into "dividing into sub-functions", and commit that tomorrow!

File: src/solver/optimisation/adequacy_patch.cpp [hunk 1]
```diff
@@ -95,28 +106,24 @@ void setBoundsAdqPatch(double& Xmax,
...
-      SetToZeroLinkNTCForAdequacyPatchFirstStep;
+    LinkCapacityForAdequacyPatchFirstStep SetToZeroLinkNTCForAdequacyPatchFirstStep;
 
     SetToZeroLinkNTCForAdequacyPatchFirstStep
-      = SetNTCForAdequacyFirstStep(
-        ProblemeHebdo->StartAreaAdequacyPatchType[Interco],
-        ProblemeHebdo->EndAreaAdequacyPatchType[Interco],
-        ProblemeHebdo->adqPatch->AdqBehaviorMap);
+      = SetNTCForAdequacyFirstStep(ProblemeHebdo->StartAreaAdequacyPatchType[Interco],
+                                   ProblemeHebdo->EndAreaAdequacyPatchType[Interco],
```
Comment (rtei-pw) [inline src/solver/optimisation/adequacy_patch.cpp]: there will be conflict with PR adq run time data, just remember to resolve in this PR or that PR

**Review by rtei-pw — APPROVED** [review]

File: src/solver/optimisation/adequacy_patch.cpp (hunk 1 above)
Comment (Milos-RTEi) [reply src/solver/optimisation/adequacy_patch.cpp]: Yes. I also realized that. Will solve it! Thanks.

**Comment by Milos-RTEi** [comment]
Best I could think of for dividing into a smaller functions. Does not make much sense dividing by SetToZero12LinksForAdequacyPatch & SetToZero11LinksForAdequacyPatch.

**Review by rtei-pw — APPROVED** [review]
I still think the original implementation is the most clear one

File: src/solver/application.cpp [hunk 2]
```diff
@@ -85,14 +85,30 @@ void checkSimplexRangeHydroHeuristic(Antares::Data::SimplexOptimization optRange
...
+        const auto& area = *(areas.byIndex[i]);
+        if (area.adequacyPatchMode == Antares::Data::AdequacyPatch::adqmPhysicalAreaInsideAdqPatch)
+        {
+            containsAdqArea = true;
+            break;
+        }
+    }
+    if (!containsAdqArea)
+        throw Error::NoAreaInsideAdqPatchMode();
+}
```
Comment (hugo-antoine-rtei) [inline src/solver/application.cpp]: You should check that adq patch is on

**Review by hugo-antoine-rtei — CHANGES_REQUESTED** [review]

File: src/solver/application.cpp (hunk 2 above)
Comment (Milos-RTEi) [reply src/solver/application.cpp]: Done.

**Review by hugo-antoine-rtei — APPROVED** [review]

**Pull request merged**